File: 0766-toeplitz-matrix/0766-toeplitz-matrix.java
class Solution {
    public boolean isToeplitzMatrix(int[][] matrix) {
        for(int i = 0; i < matrix.length; i++){
            for(int j = 0; j < matrix[0].length; j++){
                int x = i - 1, y = j - 1;
                if(0 <= x && x < matrix.length && 0 <= y && y < matrix[0].length){
                    if(matrix[x][y] != matrix[i][j]){
                        return false;
                    }
                }
            }
        }
        return true;
    }
}

File: 0766-toeplitz-matrix/0766-toeplitz-matrix.py
class Solution:
    def isToeplitzMatrix(self, matrix: List[List[int]]) -> bool:
        m, n = len(matrix), len(matrix[0])
        
        for i in range(m):
            for j in range(n):
                x, y = i - 1, j - 1
                if 0 <= x < m and 0 <= y < n:
                    if matrix[i][j] != matrix[x][y]:
                        return False
        return True